invenio_rdm_records/services/services.py

```diff
@@ -429,7 +384,15 @@ def read(self, identity, id_, expand=False, with_deleted=False):
         record = self.record_cls.pid.resolve(id_)
         result = super().read(identity, id_, expand=expand)
 
-        if record.deletion_status.is_deleted and not with_deleted:
+        can_read_deleted = self.check_permission(
```

Comment:
`with_deleted` name makes sense to me here.
I would check this permission only if `with_deleted = True`, WDTY?